datalad/support/annexrepo.py

```diff
@@ -246,15 +246,27 @@ def __init__(self, path, url=None, runner=None,
         self._batched = BatchedAnnexes(batch_size=batch_size)
 
     def __del__(self):
-        if hasattr(self, '_batched') and self._batched is not None:
-            self._batched.close()
-        super(AnnexRepo, self).__del__()
+        try:
+            if hasattr(self, '_batched') and self._batched is not None:
+                self._batched.close()
+            super(AnnexRepo, self).__del__()
+        except TypeError as e:
+            # Workaround:
+            # most likely something wasn't accessible anymore; doesn't really
+            # matter since we wanted to delete it anyway.
+            #
+            # Nevertheless, in some cases might be an issue and it is a strange
+            # thing to happen, since we check for things being None herein as
+            # well as in super class __del__;
+            # At least log it:
+            lgr.debug(exc_str(e))
+            pass
```

Comment:
Yes, we should super it up, since what `GitRepo.__del__` does is valid for `AnnexRepo` too.

`pass` is a left over indeed. Will remove it.